src/pages/actions.tsx

```diff
@@ -0,0 +1,129 @@
+import { useAppBridge } from "@saleor/app-sdk/app-bridge";
+import { Button, Text, Box } from "@saleor/macaw-ui/next";
+import { useOrdersQuery } from "../../generated/graphql";
+
+/**
+ * This is example of using AppBridge, when App is mounted in Dashboard
+ * See more about AppBridge possibilities
+ * https://github.com/saleor/saleor-app-sdk/blob/main/docs/app-bridge.md
+ *
+ * -> You can safely remove this file!
+ */
+const ActionsPage = () => {
+  const { appBridge, appBridgeState } = useAppBridge();
+  const [{ data, fetching }] = useOrdersQuery();
```

Comment:
Let's move the orders block to a separate component; this component becomes a bit too crowded to learn from. All GQL fetch handling (hook, loading state, etc) will be in one place.

I would also consider putting query inside the new component (template is already configured to do that). But that's optional.

// src/OrdersExample.tsx

```

gql`
query RecentOrders{
 orders(first: 10) {
    edges {
      node {
        id
        number
        status
        created
        user {
          firstName
          lastName
        }
      }
    }
  }
}
`

const OrdersExample = () => {
  const { appBridge, appBridgeState } = useAppBridge();
  const [{ data, fetching }] = useOrdersQuery();

  if(fetching){
    return <Text>Loading order data...</Text>
  }

  return (
  <>
      <Text color="textNeutralSubdued">
        💡 You can modify the query in the <code>graphql/queries/Orders.graphql</code> file.
        Remember to run <code>pnpm codegen</code> afterwards to regenerate the types.
      </Text>
      <ul style={{ listStyle: "inside", margin: 0 }}>
        {data?.orders?.edges?.map((order) => (
           <li key={order.node.id}>
             <a
               onClick={() => navigateToOrder(order.node.id)}
               href={`https://pikle.eu.saleor.cloud/dashboard/orders/${order.node.id}`}
                    >{`Order ${order.node.number}`}</a>{" "}
                    {`is ${order.node.status}`}
           </li>
         ))}
       </ul>
     </>
  
  )
}
```

Reply:
agree to extract component